openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -0,0 +1,179 @@
+"""
+# Add license URL
+
+Add license_url to all rows that have NULL in their meta_data fields.
+The license_url is constructed from the license and license_version fields.
+
+This is a maintenance DAG that should run once.
+"""
+import logging
+import os
+from datetime import datetime
+from textwrap import dedent
+from typing import Literal
+
+import jinja2
+from airflow.models import DAG
+from airflow.operators.python import BranchPythonOperator, PythonOperator
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from common import slack
+from common.constants import XCOM_PULL_TEMPLATE
+from common.licenses.constants import get_reverse_license_path_map
+from common.slack import send_message
+from psycopg2._json import Json
+
+
+DAG_ID = "add_license_url"
+DB_CONN_ID = os.getenv("OPENLEDGER_CONN_ID", "postgres_openledger_testing")
+
+ALERT_EMAIL_ADDRESSES = ""
+RETURN_ROW_COUNT = lambda c: c.rowcount  # noqa: E731
+
+logger = logging.getLogger(__name__)
+
+base_url = "https://creativecommons.org/"
+
+license_map = get_reverse_license_path_map()
+
+
+def get_statistics(
+    postgres_conn_id: str,
+) -> Literal["update_license_url", "final_report"]:
+    logger.info("Getting image records without license_url in meta_data.")
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    null_meta_data_count = postgres.get_first(
+        dedent(
+            """
+        SELECT COUNT(*) from image
+        WHERE meta_data IS NULL;
+        """
+        )
+    )[0]
+
+    logger.info(f"There are {null_meta_data_count} records with NULL in meta_data.")
+    if null_meta_data_count > 0:
+        return "update_license_url"
+    else:
+        return "final_report"
```

Comment:
We could potentially define these task ID names at the top (like we do with `DAG_ID`) and just reuse them throughout, just so we don't have strings floating around in several places 🙂